rl_chain/slates_chain.py

```diff
@@ -47,49 +47,107 @@ def __init__(self, actions, label=None):
         self.actions = actions
         self.label = label
 
-    @property
-    def vwtxt(self):
-        context = [f'slates shared {-self.label.r if self.label else ""} |']
+
+class SlatesTextEmbedder(base.Embedder):
+    """
+    Slates Text Embedder class that embeds the context and actions and slates into a format that can be used by VW
+    
+    Attributes:
+        embeddings_model (SentenceTransformer, optional): The type of embeddings to be used for feature representation. Defaults to BERT Sentence Transformer
+    """
+
+    def __init__(self, model: Optional[Any] = None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if model is None:
+            model = SentenceTransformer("bert-base-nli-mean-tokens")
+
+        self.model = model
+
+    def featurize(self, raw_actions: Dict[str, List[str]]):
+        # Build action embeddings
+        actions = []
+        actions_map = []
+        for (k, v) in raw_actions.items():
+            actions.append(v)
+            actions_map.append(k)
+
+        def _str(embedding):
+            return " ".join([f"{i}:{e}" for i, e in enumerate(embedding)])
+
+        action_features = [
+            [
+                _str(self.model.encode(action.impl))
+                if isinstance(action, base._Embed)
+                else action.replace(" ", "_")
+                for action in slot
+            ]
+            for slot in actions
+        ]
+
+        self.action_features = action_features
+        return actions, actions_map, action_features
+
+    def to_vw_format(self, inputs: Dict[str, Any]) -> str:
+        slates_label = inputs.get("slates_label", None)
```

Comment:
The fact that we have interface with inputs as argument but actually we are picking up only "slates_label" from it and the rest from the state variable action_features (preprocessed from the same inputs but before) seems like hack.